litestar/typing.py

```diff
@@ -442,6 +456,19 @@ def is_subclass_of(self, cl: type[Any] | tuple[type[Any], ...]) -> bool:
             if self.origin in UnionTypes:
                 return all(t.is_subclass_of(cl) for t in self.inner_types)
 
+            if isinstance(self.annotation, _GenericAlias) and self.origin not in (ClassVar, Literal):
+                cl_args = get_args(cl)
+                cl_origin = get_origin(cl) or cl
+                return (
+                    issubclass(self.origin, cl_origin)
+                    and (len(cl_args) == len(self.args) if cl_args else True)
+                    and (
+                        all(t.is_subclass_of(cl_arg) for t, cl_arg in zip(self.inner_types, cl_args))
+                        if cl_args
+                        else True
+                    )
+                )
+
```

Comment:
Trying to determine when a `_GenericAlias` type should be considered a sub-type of another type.

This says that when `A` is a `_GenericAlias` instance, then it is a subclass of `B` when:

1. if `B` is parametrized, then `A`s origin is a subtype of `B`s origin, otherwise `A`'s origin is a subtype of `B`.
2. if `B` is parametrized, then `A` and `B` must have the same number of type parameters declared
3. if `B` is parametrized, then `A`s type params are pairwise subtypes of `B`s type params

With 2 and 3, if `B` is not parametrized, and origin of `A` is a subtype of `B` then I think we should treat that as `B[Any, ...]` so the type parameters only come into it if both types have args declared.
Regarding 3, if `B` isn't parametrized, and `A`'s

Reply:
I'll have to add a lot of tests for this.